flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/TableEnvironment.scala

```diff
@@ -246,6 +251,36 @@ abstract class TableEnvironment(val config: TableConfig) {
   }
 
   /**
+    * Registers a [[ExternalCatalog]] under a unique name in the TableEnvironment's catalog.
+    * The databases and tables in the registered external catalog can be referenced in SQL queries.
+    *
+    * @param name            The name under which the externalCatalog will be registered.
+    * @param externalCatalog The externalCatalog to register.
+    */
+  def registerExternalCatalog(name: String, externalCatalog: ExternalCatalog): Unit = {
+    if (this.externalCatalogs.contains(name)) {
+      throw new ExternalCatalogAlreadyExistException(name)
+    }
+    this.externalCatalogs.put(name, externalCatalog)
+    // create an external catalog calicte schema, register it on the root schema
+    ExternalCatalogSchema.create(rootSchema, name, externalCatalog)
+  }
+
+  /**
+    * Gets a registered [[ExternalCatalog]] by name
+    *
+    * @param name The name under which the externalCatalog was previous registered.
+    * @return the externalCatalog found by name
+    */
+  def getRegisteredExternalCatalog(name: String): ExternalCatalog = {
```

Comment:
We may need to look up the registered catalog to call create/update/delete table or database method.